Composer/packages/lib/shared/src/dialogFactory.ts

```diff
@@ -146,7 +148,7 @@ export const seedNewDialog = (
       ...designerAttributes,
     },
     ...seedDefaults($type),
-    ...(initialDialogShape[$type] || {}),
+    ...(initialDialogShape[$type] ? initialDialogShape[$type](seededActions) : {}),
```

Comment:
maybe we can check the type of initial shape, such as
```
if (typeof initialShape === 'function') {
  return initialShape(...params);
} else {
  return initialShape;
}
```
Then we don't need to wrap every type with arrow function

Reply:
ah i like that